src/commands/echo.ts

```diff
@@ -0,0 +1,90 @@
+import { CommandInteraction } from "discord.js";
+import {
+  SlashCommandBuilder,
+  blockQuote,
+  bold,
+  codeBlock,
+  inlineCode,
+  italic,
+  quote,
+  spoiler,
+  strikethrough,
+  underscore,
+} from "@discordjs/builders";
+
+const formats: { [k: string]: (s: string) => string } = {
+  blockQuote,
+  bold,
+  codeBlock,
+  inlineCode,
+  italic,
+  quote,
+  spoiler,
+  strikethrough,
+  underscore,
+};
+
+export const data = new SlashCommandBuilder()
+  .setName("echo")
+  .setDescription("Replies with your input, optionally formatted")
+  .addStringOption((option) =>
+    option.setName("input").setDescription("The input to echo back").setRequired(true)
+  )
+  .addStringOption((option) =>
+    option
+      .setName("format")
+      .setDescription("The format to use")
+      .addChoices(Object.keys(formats).map((k) => [k, k]))
+  )
+  .toJSON();
+
+export const call = async (interaction: CommandInteraction) => {
+  const input = interaction.options.getString("input", true);
```

Comment:
This `true` is for `required`. Makes the return type `string` instead of `string | null`, so you don't have to check further.
Like I wrote below in the comments, it's pretty annoying to repeat this.